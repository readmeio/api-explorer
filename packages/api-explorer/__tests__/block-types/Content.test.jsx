const React = require('react');
const fs = require('fs');
const { mount } = require('enzyme');
const Content = require('../../src/block-types/Content');

test('should output one of each block type', () => {
  const body = fs.readFileSync(`${__dirname}/blocks.txt`, 'utf8');
  const content = mount(<Content body={body} isThreeColumn />);

  ['textarea', 'html', 'embed', 'api-header', 'code', 'callout', 'parameters', 'image'].forEach(
    type => {
      expect(content.find(`.magic-block-${type}`).length).toBe(1);
    },
  );
});

const body = `
  [block:api-header]
  {
    "title": "This is cool header",
    "sidebar": true
  }
  [/block]
  [block:textarea]
  {
    "text": "This is text area"
  }
  [/block]
`;

test('should output only left content if `isThreeColumn=left`', () => {
  const content = mount(<Content body={body} isThreeColumn="left" />);

  expect(content.find('.magic-block-textarea').length).toBe(1);
});

test('should output only right content if `isThreeColumn=right`', () => {
  const content = mount(<Content body={body} isThreeColumn="right" />);

  expect(content.find('.magic-block-api-header').length).toBe(1);
});

test('should make code not-dark if it is in the left column', () => {
  const content = mount(
    <Content
      body={`
        [block:code]
        {
          "codes": [
            {
              "code": "var a = 1;",
              "language": "javascript"
            }
          ]
        }
        [/block]
      `}
      isThreeColumn="left"
    />,
  );

  expect(content.html()).toContain('cm-s-neo');
});

test('should make code `dark` if it is in right column', () => {
  const content = mount(
    <Content
      body={`
        [block:code]
        {
          "codes": [
            {
              "code": "var a = 1;",
              "language": "javascript"
            }
          ],
          "sidebar": true
        }
        [/block]
      `}
      isThreeColumn="right"
    />,
  );

  expect(content.html()).toContain('cm-s-tomorrow-night');
});
