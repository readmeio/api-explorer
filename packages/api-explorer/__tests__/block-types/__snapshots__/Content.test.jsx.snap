// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the bodies of pinned callouts 1`] = `"<div class=\\"hub-reference-section\\"><div class=\\"hub-reference-left\\"><div class=\\"markdown-body\\"><div class=\\"pin\\"><blockquote class=\\"callout callout_warn\\" theme=\\"ðŸš§\\"><h3 class=\\"callout-heading false\\"><span class=\\"callout-icon\\">ðŸš§</span><p>covfefe</p></h3><p>This body should be rendered twice!</p></blockquote></div></div></div><div class=\\"hub-reference-right\\"><div class=\\"markdown-body\\"><div class=\\"pin\\"><blockquote class=\\"callout callout_warn\\" theme=\\"ðŸš§\\"><h3 class=\\"callout-heading false\\"><span class=\\"callout-icon\\">ðŸš§</span><p>covfefe</p></h3><p>This body should be rendered twice!</p></blockquote></div></div></div></div>"`;
